File: Đồ án OOP C# - Quản lý phòng bán vé xe khách/DoAn2/CapnhatDoanhthuvasove.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DoAn2
{
    interface CapnhatDoanhthuvasove
    {
        void Capnhatsaukhixoa(int machuyen, double giave);
        void Suasovedabanvadoanhthu(Ngoaithanh sua, int temp);
        int Searchmachuyen(int machuyen);
        Ngoaithanh ReturnNgoaithanh(int temp);
    }

}

File: Đồ án OOP Java - Quản lý phòng vé bán xe khách/src/Doan2/CapnhatDoanhthuvasove.java
package Doan2;

import java.io.FileNotFoundException;
import java.io.IOException;

public interface CapnhatDoanhthuvasove
{		// interface này bổ sung thêm những phương thức đặc trưng phụ vụ cho các thao tác cho class Ngoại thành 
		//mà class Nội thành không có
		void Capnhatsaukhixoa(int machuyen, double giave) throws FileNotFoundException, IOException ;
		void Suasovedabanvadoanhthu(Ngoaithanh sua, int temp);
		int Searchmachuyen(int machuyen);
		Ngoaithanh ReturnNgoaithanh(int temp);
}
